pypesto/petab/importer.py

```diff
@@ -171,11 +166,21 @@ def __init__(
             model_name = _find_model_name(self.output_folder)
         self.model_name = model_name
 
+        self.simulator_type = simulator_type
+        self.simulator = simulator
```

Comment:
As the importer only needs to use this in `create_objective`, why not remove these attributes and implement them as arguments of `create_objective` instead?

e.g. if I want to compare the libroadrunner and AMICI objectives, why do I need two importer objects? I could use the same importer like `importer.create_objective_creator(..., simulator_id=AMICI)` and `importer.create_objective_creator(..., simulator_id=ROADRUNNER)`

No strong opinion on this though, fine as is

Reply:
+1 
Think this makes sense. Also not strong opinion.

Reply:
having this would make it imo much easier to 
1. init the importer -> 2. importer.create_problem()
without having to pass these arguments constantly to create_problem. Could add it as additional possibility to `create_objective` such that people can do either? Argument in `create_objective` would then take precedence

Reply:
IMO the cleanest implementation would be that default behavior (AMICI simulator) is usable as you said like
```python
importer = PetabImporter(petab_problem)
problem = importer.create_problem()
```

but custom behavior is like
```python
importer = PetabImporter(petab_problem)
obj = importer.create_objective(AMICI)
obj.amici_solver.setAbsoluteTolerance(1e-8)
problem = importer.create_problem(obj)
```
Since it's quite common for me to customize things like simulator tolerances, passing `obj` to `create_problem` is no change for me.

Just to explain my (weak) preference.

Reply:
passing the objective to problem is still possible. And the other behavior would look like (+- details)
```python
importer = PetabImporter(petab_problem, AMICI)
obj = importer.create_objective()
obj.amici_solver.setAbsoluteTolerance(1e-8)
problem = importer.create_problem(obj)
```
I think going with allowing both would be best solution to be able to adjust to individual preferences?

Reply:
By now, `PetabImporter` doesn't have much functionality or internal state left. Maybe we could directly go for something like

```python
importer = AmiciPetabImporter(petab_problem, ...)
problem = importer.create_problem()
...
```

Reply:
> By now, `PetabImporter` doesn't have much functionality or internal state left. Maybe we could directly go for something like
> 
> ```python
> importer = AmiciPetabImporter(petab_problem, ...)
> problem = importer.create_problem()
> ...
> ```

I think currently the main advantage is that we can easily switch by just changing an argument in the Importer to switch simulators if needed/wanted which we would lose with the above. additionally the `create_problem` function is for all of them kind of the same which just in my mind creates this natural hierarchy of "Importer -> Inner thing (=Objective creator)"